Assets/Scripts/NPCs/BasicNurse.cs

```diff
@@ -197,7 +197,24 @@ private void ToggleAnimations(bool toggle)
             // I was thinking that toggle=false just turns off all animations (disable the animator?)
             // and toggle=true can do specific behaivor based on currentState
```

Comment:
There's only one issue; I set currentState to the new state only at the **end** of the SetState() method.  Because of this, when you check currentState in ToggleAnimations(), you're actually checking the _previous_ state instead of the one that you're about to change to.  I'll send you a modified version of BasicNurse.cs that fixes this so that ToggleAnimations can work as-is.